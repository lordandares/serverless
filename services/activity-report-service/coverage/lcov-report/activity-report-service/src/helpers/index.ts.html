<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for activity-report-service/src/helpers/index.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">activity-report-service/src/helpers</a> index.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>34/34</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>18/18</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as s3 from './s3'
&nbsp;
export { buildLocationSummaryPdf } from './build-location-summary-pdf'
export { buildShiftSummaryPdf } from './build-shift-summary-pdf'
export { buildEmail } from './build-email'
export { buildShiftEmail } from './build-shift-email'
export { getShiftDetails } from './get-shift-details'
export { getShiftReportLinks } from './get-shift-report-links'
export { getUserDetails } from './get-user-details'
export { concatPdfs } from './concat-pdfs'
export { downloadFiles } from './download-files'
export { fetchApplications } from './fetch-applications'
export { fetchLocations } from './fetch-locations'
export { getLocationReportLinks } from './get-location-report-links'
export { getShiftSummaryData } from './get-shift-summary-data'
export { getSummaryData } from './get-summary-data'
export { getFormS3Paths } from './get-references'
export { s3 }
export { sendEmail } from './send-email'
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jan 13 2020 21:52:44 GMT+0000 (Coordinated Universal Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
