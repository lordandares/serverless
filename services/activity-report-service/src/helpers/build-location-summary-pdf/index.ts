import { buildActivityPdf } from '@lighthouse/common'
import * as dateFns from 'date-fns'

import { buildConfig, createSummaryPdf } from '../create-summary-pdf'

export async function buildLocationSummaryPdf(data) {
  const config = buildConfig(data)
  const { data: summaryData, pdfOptions } = config

  const pdfDefinition = await buildActivityPdf(pdfOptions, summaryData)
  const s3FullPath = getLocationSummaryPdfPath(data)
  await createSummaryPdf(pdfDefinition, s3FullPath)

  return s3FullPath
}

function getLocationSummaryPdfPath({ location }) {
  const date = new Date()
  const year = dateFns.format(date, 'YYYY')
  const month = dateFns.format(date, 'MM')
  const day = dateFns.format(date, 'DD')
  const unix = date.getTime()

  const { _id: locationId } = location
  const s3PathName = `activity-reports/${locationId}/${year}/${month}`
  const s3FileName = `${year}-${month}-${day}-summary-${unix}.pdf`
  return `${s3PathName}/${s3FileName}`
}
