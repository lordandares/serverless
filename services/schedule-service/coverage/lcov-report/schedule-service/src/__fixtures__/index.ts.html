<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for schedule-service/src/__fixtures__/index.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">schedule-service/src/__fixtures__</a> index.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145</td><td class="line-coverage quiet"><span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const scheduleDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    areas: ['area1'],
    enabled: true,
    locations: ['location1'],
    name: 'Schedule One',
    included: {},
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
    strategy: {
      options: {
        duration: {
          unit: 'minute',
          value: 15,
        },
        frequency: {
          unit: 'minute',
          value: 15,
        },
      },
      type: 'stopwatch',
    },
    type: 'visit',
  },
  endAt: '2019-10-1500:00:00.000Z',
  groupType: 'schedule',
  pk: 'application1-schedule',
  scheduleId: 'schedule1',
  sk: '2019-10-13T00:00:00.000Z-schedule',
  startAt: '2019-10-1400:00:00.000Z',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}
&nbsp;
export const schedulePayload = {
  areas: ['area1'],
  enabled: true,
  endAt: '2019-10-1500:00:00.000Z',
  locations: ['location1'],
  name: 'Schedule One',
  serviceHours: {
    hours: [
      {
        description: 'MON 11:00 - TUE 00:00',
        end: 86400000,
        start: 39600000,
        type: 'DEFAULT',
      },
    ],
    timezone: 'Australia/Melbourne',
  },
  startAt: '2019-10-1400:00:00.000Z',
  strategy: {
    options: {
      duration: {
        unit: 'minute',
        value: 15,
      },
      frequency: {
        unit: 'minute',
        value: 15,
      },
    },
    type: 'stopwatch',
  },
  type: 'visit',
}
&nbsp;
export const occurrenceDocument = {
  applicationId: 'application1',
  createdAt: '2019-01-01T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  endAt: '2019-02-02T00:00:00.000Z',
  groupType: 'occurrence',
  location: 'location1',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-01-01T00:00-occurrence',
  startAt: '2019-02-01T00:00:00.000Z',
  type: 'task',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}
&nbsp;
export const areaDocument = {
  _id: 'location1',
  application: '56789671d4c6488b128c7f9c',
  name: 'WRNMMC Facility',
  altName: 'WRNMMC Facility',
  type: 'location',
  tags: [],
  address: {
    street: '8901 Rockville Pike',
    street2: '',
    city: 'Bethesda',
    state: 'MD',
    postal: '20889',
    country: 'US',
  },
  keywords: ['hospital'],
  areaSize: 536559.55514508,
  serviceHours: {
    hours: [
      {
        description: 'MON 00:00 - TUE 00:00',
        duration: 86400000,
        end: 86400000,
        start: 0,
        type: 'DEFAULT',
      },
    ],
    timezone: 'America/New_York',
  },
  deleted: false,
  timezone: 'America/New_York',
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 25 2019 18:46:45 GMT+0000 (Coordinated Universal Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
