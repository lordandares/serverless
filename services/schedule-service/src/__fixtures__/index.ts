import { Context } from 'aws-lambda'

export const areaDocument = {
  _id: 'location1',
  address: {
    city: 'Bethesda',
    country: 'US',
    postal: '20889',
    state: 'MD',
    street: '8901 Rockville Pike',
    street2: '',
  },
  altName: 'WRNMMC Facility',
  application: '56789671d4c6488b128c7f9c',
  areaSize: 536559.55514508,
  deleted: false,
  floorsRef: [0],
  groupType: 'location',
  keywords: ['hospital'],
  name: 'WRNMMC Facility',
  serviceHours: {
    hours: [
      {
        description: 'MON 00:00 - TUE 00:00',
        duration: 86400000,
        end: 86400000,
        start: 0,
        type: 'DEFAULT',
      },
    ],
    timezone: 'America/New_York',
  },
  tags: [],
  timezone: 'America/New_York',
}

export const occurrenceDocument = {
  applicationId: 'application1',
  createdAt: '2019-01-01T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    occurrenceInterval: [1, 2],
    scheduleName: 'Schedule 1',
    serviceInterval: [1, 2],
    timezone: 'Australia/Melbourne',
    type: 'visit',
  },
  endAt: '2019-02-01T23:59:59.999Z',
  groupType: 'occurrence',
  locationId: '5d163137b8b3b7000127edd1',
  location_endAt_occurrenceId: 'location1-2019-02-02T00:00:00.000Z-occurrence1',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-01-01T00:00-occurrence1',
  startAt: '2019-02-01T00:00:00.000Z',
  status: 'pending',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}

export const occurrenceTimerDocument = {
  pk: `timer-2019-01-01T00:00`,
  sk: `timer#application1-occurrence#2019-01-01T00:00-occurrence`,
  endAt: '2019-01-01T23:59:59.999Z',
  groupType: `occurrence-timer`,
  scheduleId: 'schedule1',
  targetArn: 'someArn',
}

export const scheduleDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    areas: ['area1'],
    enabled: true,
    included: {},
    locations: ['location1'],
    name: 'Schedule One',
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
    strategy: {
      options: {
        duration: {
          unit: 'minute',
          value: 15,
        },
        frequency: {
          unit: 'minute',
          value: 15,
        },
      },
      type: 'stopwatch',
    },
    type: 'visit',
  },
  endAt: '2019-10-1500:00:00.000Z',
  groupType: 'schedule',
  pk: 'application1-schedule',
  scheduleId: 'schedule1',
  sk: '2019-10-13T00:00:00.000Z-schedule1',
  startAt: '2019-10-1400:00:00.000Z',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}

export const scheduleLocationDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: '123456789',
    label: 'label',
    type: 'system',
  },
  data: {
    name: 'Schedule One',
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
  },
  groupType: 'location',
  pk: 'application1-location',
  schedules: ['schedule1'],
  sk: 'location1',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: '123456789',
    label: 'label',
    type: 'system',
  },
}

export const schedulePayload = {
  areas: ['area1'],
  enabled: true,
  endAt: '2019-10-1500:00:00.000Z',
  locations: ['location1'],
  name: 'Schedule One',
  serviceHours: {
    hours: [
      {
        description: 'MON 11:00 - TUE 00:00',
        end: 86400000,
        start: 39600000,
        type: 'DEFAULT',
      },
    ],
    timezone: 'Australia/Melbourne',
  },
  startAt: '2019-10-1400:00:00.000Z',
  strategy: {
    options: {
      duration: {
        unit: 'minute',
        value: 15,
      },
      frequency: {
        unit: 'minute',
        value: 15,
      },
    },
    type: 'stopwatch',
  },
  type: 'visit',
}

export const rulePayload = {
  applicationId: 'application-id-1111',
  data: {
    occurrenceInterval: [1580761454998, 1580762054997],
  },
  hook: {
    endpoint: 'sns:endpoint:uri',
    type: 'sns',
  },
  locationId: 'location-id-11111',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-06-02T00:00:00.000Z-occurrence1',
  startAt: '2019-02-01T00:00:00.000Z',
  type: 'visit',
}

export const rulePatternDocument = {
  createdAt: '2019-06-02T00:00:00.000Z',
  createdBy: {
    id: 'created-by-id-11111',
    label: 'created-by-label-1111',
    type: 'created-by-type-11111',
  },
  groupType: 'rule-pattern',
  matches: {
    occurrence1: {
      pk: 'application1-occurrence',
      sk: '2019-06-01T00:00:00.000Z-occurrence1',
    },
  },
  pk: 'rule-pattern-application-id-11111-location-id-11111',
  sk: 'visit',
  updatedAt: '2019-06-02T00:00:00.000Z',
}

export const context: Context = {
  awsRequestId: 'aws-request-id',
  callbackWaitsForEmptyEventLoop: false,
  clientContext: undefined,
  done: jest.fn(),
  fail: jest.fn(),
  functionName: 'function-name',
  functionVersion: 'function-version',
  getRemainingTimeInMillis: jest.fn(),
  identity: undefined,
  invokedFunctionArn: 'invoked-function-arn',
  logGroupName: 'log-group-name',
  logStreamName: 'log-stream-name',
  memoryLimitInMB: 0,
  succeed: jest.fn(),
}
