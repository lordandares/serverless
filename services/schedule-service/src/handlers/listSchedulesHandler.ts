import { errors, http, schemas } from '@lighthouse/serverless-common'
import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'
import { ScheduleService } from '../service/ScheduleService'

export async function listSchedulesHandler(
  event: APIGatewayProxyEvent,
): Promise<APIGatewayProxyResult> {
  try {
    const { headers } = event

    if (!headers) {
      throw new errors.ApplicationError({
        message: 'ListSchedulesHandlers: missing headers on event',
      })
    }

    const applicationId: string = headers['lio-application-id']
    const schedules: schemas.SchedulePayloadSchema[] = await ScheduleService.listSchedules(
      {
        applicationId,
      },
    )

    return {
      body: JSON.stringify(schedules),
      statusCode: http.STATUS_CODES.SUCCESSFUL.OK,
    }
  } catch (err) {
    console.error('ListSchedulesError', {
      err,
      event,
    })

    return errors.httpErrorHandler(err)
  }
}
