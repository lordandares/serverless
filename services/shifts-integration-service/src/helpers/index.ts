export { default as getPlatformEndpoint } from './get-platform-endpoint'
export { default as getSnsPayload } from './get-sns-payload'
export {
  getWinTeamMessages,
  getWinTeamTimePunch,
} from './get-winteam-time-punch'
export { default as notifyUser } from './notify-user'
export { default as parseWinTeamMessages } from './parse-winteam-messages'
export { default as sendMessage } from './send-message'
export {
  default as validateEndpointAttributes,
} from './validate-endpoint-attributes'
