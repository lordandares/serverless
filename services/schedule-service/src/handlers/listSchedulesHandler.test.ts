import { schemas } from '@lighthouse/serverless-common'
import * as AWS from 'aws-sdk-mock'

import { createApiEvent } from '../../../../__test__/helpers'
import { scheduleDocument } from '../__fixtures__'
import { listSchedulesHandler } from './listSchedulesHandler'

beforeEach(() => (process.env.TABLE_SCHEDULES = 'table-schedules'))

test('event headers error', async () => {
  expect.assertions(1)

  const result = await listSchedulesHandler(createApiEvent({ headers: null }))

  expect(result).toMatchInlineSnapshot(`
Object {
  "body": "{\\"name\\":\\"ApplicationError\\",\\"status\\":500,\\"message\\":\\"ListSchedulesHandlers: missing headers on event\\"}",
  "statusCode": 500,
}
`)
})

test('event applicationId header error', async () => {
  expect.assertions(1)

  const result = await listSchedulesHandler(
    createApiEvent({
      headers: { 'lio-application-id': '' },
    }),
  )

  expect(result).toMatchInlineSnapshot(`
Object {
  "body": "{\\"name\\":\\"ApplicationError\\",\\"status\\":500,\\"message\\":\\"The \`applicationId\` is missing from headers\\"}",
  "statusCode": 500,
}
`)
})

test('generic error', async () => {
  expect.assertions(2)

  const querySpy = jest.fn().mockRejectedValue(new Error('NetworkError'))

  AWS.mock('DynamoDB.DocumentClient', 'query', querySpy)

  const result = await listSchedulesHandler(
    createApiEvent({ headers: { 'lio-application-id': 'application1' } }),
  )

  expect(querySpy).toHaveBeenCalled()
  expect(result).toMatchInlineSnapshot(`
Object {
  "body": "{\\"name\\":\\"ApplicationError\\",\\"status\\":500,\\"message\\":\\"NetworkError\\"}",
  "statusCode": 500,
}
`)

  AWS.restore()
})

test('Returns 200 for existing documents', async () => {
  expect.assertions(3)

  const querySpy = jest.fn().mockResolvedValue({
    Count: 1,
    Items: [{ ...scheduleDocument }],
  })

  AWS.mock('DynamoDB.DocumentClient', 'query', querySpy)

  const result = await listSchedulesHandler(
    createApiEvent({ headers: { 'lio-application-id': 'application1' } }),
  )

  expect(querySpy).toHaveBeenCalled()
  expect(querySpy.mock.calls).toMatchSnapshot()
  expect(result).toMatchSnapshot()

  AWS.restore()
})
