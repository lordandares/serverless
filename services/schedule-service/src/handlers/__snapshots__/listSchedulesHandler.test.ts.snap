// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns 200 for existing documents 1`] = `
Array [
  Array [
    Object {
      "ExpressionAttributeValues": Object {
        ":groupType": "schedule",
        ":pk": "application1-schedule",
      },
      "FilterExpression": "groupType = :groupType",
      "KeyConditionExpression": "pk = :pk",
      "ScanIndexForward": false,
      "TableName": "table-schedules",
    },
    [Function],
  ],
]
`;

exports[`Returns 200 for existing documents 2`] = `
Object {
  "body": "[{\\"areas\\":[\\"area1\\"],\\"createdAt\\":\\"2019-10-13T00:00:00.000Z\\",\\"createdBy\\":{\\"id\\":\\"user1\\",\\"label\\":\\"Unknown User\\",\\"type\\":\\"user\\"},\\"enabled\\":true,\\"endAt\\":\\"2019-10-1500:00:00.000Z\\",\\"id\\":\\"schedule1\\",\\"included\\":{},\\"locations\\":[\\"location1\\"],\\"name\\":\\"Schedule One\\",\\"serviceHours\\":{\\"hours\\":[{\\"description\\":\\"MON 11:00 - TUE 00:00\\",\\"end\\":86400000,\\"start\\":39600000,\\"type\\":\\"DEFAULT\\"}],\\"timezone\\":\\"Australia/Melbourne\\"},\\"startAt\\":\\"2019-10-1400:00:00.000Z\\",\\"strategy\\":{\\"options\\":{\\"duration\\":{\\"unit\\":\\"minute\\",\\"value\\":15},\\"frequency\\":{\\"unit\\":\\"minute\\",\\"value\\":15}},\\"type\\":\\"stopwatch\\"},\\"type\\":\\"visit\\",\\"updatedAt\\":\\"2019-10-13T01:00:00.000Z\\",\\"updatedBy\\":{\\"id\\":\\"user1\\",\\"label\\":\\"Unknown User\\",\\"type\\":\\"user\\"}}]",
  "statusCode": 200,
}
`;
